api/proto/teleport/legacy/client/proto/joinservice.proto

```diff
@@ -158,6 +158,32 @@ message TPMEncryptedCredential {
   bytes secret = 2;
 }
 
+// RegisterUsingOracleMethod request is the request for registration via the
+// Oracle join method.
+message RegisterUsingOracleMethodRequest {
```

Comment:
What will be the order of flow here? e.g does client send first or the server? which messages?

I think it'd be beneficial for us to ensure that RegisterUsingToken request is sent first if possible - this ensures that the Auth Server knows a bunch of key details if the flow fails part-way through. One of the problems we have with the IAM method at the moment is that we have basically zero details about the failing bot/agent if a failure occurs during the join process because all the details are sent at the end.

Reply:
👍 RegisterUsingToken is now sent first